### PR TITLE
Lua API: fix OOB array access in find_nodes_near

### LINKED ISSUE #14946: Minetest segfaults in minetest.find_nodes_near() with non-existent group
### Minetest version

```true
Minetest 5.9.0-dev-7625f88
Also affects 5.7.0, and probably some more
```

### Irrlicht device

_No response_

### Operating system and version

Devuan Daedalus (eq to Debian Bookworm)

### CPU model

_No response_

### GPU model

_No response_

### Active renderer

_No response_

### Summary

When minetest.find_nodes_in_area() is called with a "group:" name that hasn't been used anywhere yet, and grouped = true, it causes Minetest to segfault.
 I discovered this when I made a node that searched for another by group, but forgot to actually add the group to the target. You can also get hit by this if you have a typo in a group name. It suffices that the value of "group:*" has not been used by any nodes.

### Steps to reproduce

`minetest.find_nodes_in_area(pos, pos, "group:invalidgroupname", true)`
 is all you need to do it

### PR DESCRIPTION
**Pull request by SmallJoker**
Fixes #14946. Thanks to sfan5 for the backtrace.

## To do

This PR is Ready for Review.

## How to test

```Lua
minetest.register_chatcommand("p", {
	func = function(name, param)
		local pos = minetest.get_player_by_name(name):get_pos()
		local minp = vector.add(pos, -5)
		local maxp = vector.add(pos,  5)

		--local groupname = "group:stone"
		local groupname = "group:invalidgroupname"
		local names_positions = minetest.find_nodes_in_area(minp, maxp, groupname, true)
		for name, pos in pairs(names_positions) do
			print(name, #pos)
		end
		local positions, counts = minetest.find_nodes_in_area(minp, maxp, groupname, false)
		print(#positions, dump(counts))
		return true, "OK!"
	end
})
```

master: segfault
PR: empty table(s)

Re-tested with `group:stone` and got the identical output for master and diff.

### CODE REVIEW

File: src/script/lua_api/l_env.cpp [hunk 1]
```diff
@@ -970,16 +970,16 @@ int ModApiEnvBase::findNodesInArea(lua_State *L, const NodeDefManager *ndef,
 		});
 
 		// last filter table is at top of stack
-		u32 i = filter.size() - 1;
-		do {
+		s32 i = filter.size();
+		while (i --> 0) {
```
Comment (sfan5) [inline src/script/lua_api/l_env.cpp:974]: shouldn't `i-- != 0` work without needing to change the type?

Comment (SmallJoker) [reply src/script/lua_api/l_env.cpp:974]: You're right. I in fact tested the first version with `u32` and - for whatever reason - thought that I need to have it signed if `i = 0`, which is not true.

Comment (SmallJoker) [reply src/script/lua_api/l_env.cpp:974]: You're right. I in fact tested the first version with `u32` and - for whatever reason - thought that I need to have it signed if `i = 0`, which is not necessary.

**Comment by jeremyshannon** [comment]
Could it perhaps issue a warning that no such group exists?

**Comment by cx384** [comment]
Works on my machine.

**Pull request merged**